src/components/Profile.js

```diff
@@ -57,35 +54,26 @@ class Profile extends Component<Props> {
       && !this.props.trustedByUids.has(this.props.authMemberDoc.id);
   }
 
-  isOwnProfile(uid) {
-    return this.props.authMemberDoc && this.props.authMemberDoc.uid === uid;
+  isOwnProfile() {
+    const { authFirebaseUser, memberDoc } = this.props;
+    return authFirebaseUser && memberDoc && authFirebaseUser.uid === memberDoc.id;
   }
 
-
   render() {
-    const { authFirebaseUser, authIsLoaded, isMyProfile, memberDoc } = this.props;
-    if (!authIsLoaded) {
-      return <Loading />;
-    }
-    if (isMyProfile && !authFirebaseUser) {
-      return (
-        <div>
-          <span>You need to </span>
-          <Redirect to="/login" />
-        </div>
-      );
-    }
-    if (!memberDoc) {
+    const { authFirebaseUser, authIsLoaded, memberDoc } = this.props;
+    if (!authIsLoaded || !memberDoc) {
       return <Loading />;
     }
+    const ownProfile = this.isOwnProfile();
     if (!memberDoc || !memberDoc.get('mid')) {
```

Comment:
I don't think we need `!memberDoc` here since we return on that above.

Reply:
Nah he deleted that; if `authIsLoaded === false`, then `memberDoc` could still be available.